ui/pages/destinations/add.js

```diff
@@ -238,11 +238,28 @@ export default function DestinationsAdd() {
               </h3>
             </div>
           )}
-          <Link href='/destinations'>
-            <a className='flex-none items-center self-center rounded-md border border-transparent bg-black px-4 py-2 text-2xs font-medium text-white shadow-sm hover:bg-gray-800'>
-              Finish
-            </a>
-          </Link>
+          <button
+            className='inline-flex items-center rounded-md border border-transparent bg-black px-4 py-2 text-2xs font-medium text-white shadow-sm hover:cursor-pointer hover:bg-gray-800 disabled:cursor-not-allowed disabled:opacity-30'
+            type='button'
+            onClick={async () => {
+              // grant the person that added this cluster 'cluster-admin' access automatically
```

Comment:
I don't think it's safe to assume the individual adding the cluster should automatically get any grants, particularly `cluster-admin`.  It's possible this elevates their privileges beyond what they already had. I suggest not merging this change.

Reply:
If they're an admin in Infra then they could just grant themselves cluster admin access after adding the cluster, so this is not a privilege escalation.